src/theme/SearchBar/FilterDropdown.js

```diff
@@ -0,0 +1,168 @@
+
+      if (!checked && updatedFacetList.length === 0) {
+        return;
+      }
+
+      if (checked) {
+        // If a parent filter is checked, add its child filters
+        if (value === 'docusaurus_tag:docs-build-current') {
+          updatedFacetList = [
+            ...new Set([...updatedFacetList, ...buildSubFilters]),
+          ];
+        } else if (value === 'docusaurus_tag:docs-identity-rs-1-0-current') {
+          updatedFacetList = [
+            ...new Set([...updatedFacetList, ...identitySubFilters]),
+          ];
+        } else if (value === 'docusaurus_tag:docs-maintain-current') {
+          updatedFacetList = [
+            ...new Set([...updatedFacetList, ...maintainSubFilters]),
+          ];
+        }
+      } else {
+        // If a parent filter is unchecked, remove its child filters
+        if (value === 'docusaurus_tag:docs-build-current') {
+          updatedFacetList = updatedFacetList.filter(
+            (facet) => !buildSubFilters.includes(facet),
+          );
+        } else if (value === 'docusaurus_tag:docs-identity-rs-1-0-current') {
+          updatedFacetList = updatedFacetList.filter(
+            (facet) => !identitySubFilters.includes(facet),
+          );
+        } else if (value === 'docusaurus_tag:docs-maintain-current') {
+          updatedFacetList = updatedFacetList.filter(
+            (facet) => !maintainSubFilters.includes(facet),
+          );
+        }
+      }
+
+      setSelectedFacets([selectedFacets[0], updatedFacetList]);
+    };
+
+    const handleSelectAll = (event) => {
+      const checked = event.target.checked;
+      const updatedFacetList = checked ? actualFacets.slice() : [];
+      setSelectedFacets([selectedFacets[0], updatedFacetList]);
+    };
+
+    useEffect(() => {
+      // Check if all subfilters are selected, if yes, select the parent filter
+      const parentFilterStatus = {
+        'docusaurus_tag:docs-maintain-current': maintainSubFilters.every(
+          (facet) => selectedFacets[1].includes(facet),
+        ),
+        'docusaurus_tag:docs-build-current': buildSubFilters.every((facet) =>
+          selectedFacets[1].includes(facet),
+        ),
+        'docusaurus_tag:docs-identity-rs-1-0-current': identitySubFilters.every(
+          (facet) => selectedFacets[1].includes(facet),
+        ),
+      };
+
+      Object.keys(parentFilterStatus).forEach((parentFilter) => {
+        if (
+          parentFilterStatus[parentFilter] &&
+          !selectedFacets[1].includes(parentFilter)
+        ) {
+          setSelectedFacets((prevState) => [
+            prevState[0],
+            [...prevState[1], parentFilter],
+          ]);
+        }
+      });
+    }, [selectedFacets, setSelectedFacets]);
+
+    useEffect(() => {
+      const handleScroll = () => {
+        if (window.scrollY > 28) {
+          setIsScrolled(true);
+        }
+      };
+      handleScroll();
+    }, [window.scrollY]);
```

Comment:
I think this is the source of the build failure. This dropdown is used on the search page, which means it will be rendered during static site generation. `window.scrollY` is used here as a dependency, but `window` is only available client-side, not server-side. See here how to resolve this: https://docusaurus.io/docs/advanced/ssg#escape-hatches

Reply:
Thank you for the help above link, this issue is now resolved.